### PR TITLE
Improved lookup_name_gender function

### PR DESCRIPTION
**Pull request by sveinbjornt**
lookup_name_gender function now defaults to preferring name meanings in the nominative case (to prevent lookup of e.g. "Gunnar" returning "kvk" due to the gen. case of "Gunnur"). This bug was caught by our query system integration tests which is nice! They're proving their worth, especially when we upgrade BÍN data and surprising new edge cases crop up.

### CODE REVIEW

**Review by sultur — APPROVED** [review]
lgtm! We discussed possibly using last name endings ("-son"/"-dóttir") to help when disambiguation is difficult such as in the case of the names Auður or Blær. Do we want to add that to this PR or add that later? This PR at least fixes lookups for a bunch of names.

**Comment by sultur** [comment]
image
The tests fail due to this :joy:.

File: src/reynir/bindb.py [hunk 1]
```diff
@@ -108,16 +108,27 @@ def meanings(self, w: str) -> List[BIN_Tuple]:
         ]
 
     @lru_cache(maxsize=_NAME_GENDER_CACHE_SIZE)
-    def lookup_name_gender(self, name: str) -> str:
+    def lookup_name_gender(self, name: str, preferred_case: str = "nf") -> str:
```
Comment (vthorsteinsson) [inline src/reynir/bindb.py:111]: Perhaps add * to force keyword-only parameter? i.e. `(self, name: str, *, preferred_case: str = "nf")`

**Review by vthorsteinsson — COMMENTED** [review]
LGTM except for suggestion on keyword-only parameter.
